chain/client/src/gc_actor.rs

```diff
@@ -43,20 +48,27 @@ impl GCActor {
             runtime_adapter,
             gc_config,
             epoch_manager,
+            shard_tracker,
+            validator_signer,
             is_archive,
             no_gc: false,
         }
     }
 
     fn clear_data(&mut self) -> Result<(), near_chain::Error> {
+        let signer = self.validator_signer.get();
+        let me = signer.as_ref().map(|signer| signer.validator_id());
         // A RPC node should do regular garbage collection.
         if !self.is_archive {
             return self.store.clear_data(
                 &self.gc_config,
                 self.runtime_adapter.clone(),
                 self.epoch_manager.clone(),
+                &self.shard_tracker,
+                me,
             );
         }
+        debug_assert!(self.shard_tracker.tracks_all_shards());
```

Comment:
What's the purpose of this assert? Will something break into state cleanup if one day archival nodes don't track all shards?

Reply:
One thing I had in mind is that hot db could potentially do some cleanup and remove the mapping, while the mapping would still be used in cold db.

Reply:
Added a comment in https://github.com/near/nearcore/pull/12734/commits/6b191d937316e2228a7991b2748c5b963ce71265